src/TextInputWrapper.cpp

```diff
@@ -0,0 +1,100 @@
+#include <Gosu/Gosu.hpp>
+
+namespace Gosu
+{
+  class TextInputForWrapper : public Gosu::TextInput
+  {
+    public:
+      TextInputForWrapper();
+      std::string filter(std::string text) const override;
+      std::function<void (const char* text)> filter_callback;
+      std::string filter_result = "";
+  };
+}
+
+Gosu::TextInputForWrapper::TextInputForWrapper() : Gosu::TextInput()
+{
+}
+std::string Gosu::TextInputForWrapper::filter(std::string text) const
+{
+  if (filter_callback != nullptr) {
+    filter_callback(text.c_str());
+    return filter_result;
+  }
+  else {
+    return text;
+  }
+}
+
+
+extern "C" {
+  #include <Gosu/TextInput.h>
+
+  Gosu_TextInput* Gosu_TextInput_create()
+  {
+    return reinterpret_cast<Gosu_TextInput*>( new Gosu::TextInputForWrapper() );
+  }
+
+  const char* Gosu_TextInput_text(Gosu_TextInput* text_input)
+  {
+    static thread_local std::string string;
```

Comment:
`static thread_local` is redundant: https://stackoverflow.com/a/22794640

Can you please remove the `static`?